#ifndef MAINWINDOW_H
#define MAINWINDOW_H
#include "logwidget.h"
#include <QMainWindow>

/*------------------------ç”¨æˆ·ä»£ç å¤´æ–‡ä»¶---------------------------*/
#include <QtSerialPort/QSerialPort>//ä¸²å£
#include <QtSerialPort/QSerialPortInfo>//ä¸²å£
#include <QDebug>//ç”¨äºåœ¨æ§åˆ¶å°è¾“å‡ºè°ƒè¯•ä¿¡æ¯
#include <QTime>//å®šæ—¶å™¨
#include <QPainter>//åæ ‡ç³»ç»˜å›¾

#include <QSqlDatabase>
#include <QSqlQuery>
#include <QSqlError>
#include <QProcess>
#include <QShowEvent>

QT_BEGIN_NAMESPACE
namespace Ui {
class MainWindow;
}
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private slots:
    void on_openSerialBtn_clicked();
    void on_scanSerialBtn_clicked();
    void AnalyzeData();           // æ•°æ®è¯»å–
    void setupPlot();             // ç»˜å›¾åˆå§‹åŒ–
    void InitDatabase();          // åˆå§‹åŒ–æ•°æ®åº“
    void LoadDataToTable();       // åŠ è½½æ•°æ®åº“åˆ°è¡¨æ ¼

    void on_pushButton_clicked(); // å¯åŠ¨ Python æ¨èé€Ÿåº¦
    void on_pushButton_2_clicked(); // äººè„¸è¯†åˆ«æŒ‰é’®
    void on_addWeight_clicked();  // æ·»åŠ é‡é‡è„šæœ¬
    void on_avgweight_clicked();  // å†™å¹³å‡é‡é‡

protected:
    void showEvent(QShowEvent *event) override;

private:
    Ui::MainWindow *ui;

    LogWidget *m_log;

    int serialflag;

    QSerialPort *myserial;        // ä¸²å£å¯¹è±¡
    bool serial_flag, start_flag; // ä¸²å£æ ‡å¿—ã€å¯åŠ¨æ ‡å¿—
    QByteArray alldata;           // ä¸²å£æ€»æ•°æ®ç¼“å­˜
    QByteArray m_receiveBuffer;   // åˆ†æç”¨ç¼“å†²åŒº

    QDateTime mycurrenttime;      // å½“å‰æ—¶é—´
    QDateTime mystarttime;        // å¼€å§‹æ—¶é—´

    QProcess *faceProcess;

    bool allow_db_insert_flag;    // ğŸ”¥ æ–°å¢æ ‡å¿—ï¼šæ˜¯å¦å…è®¸æ’å…¥æ•°æ®åº“
};

#endif // MAINWINDOW_H
